hooks/useAuthentication.js

```diff
@@ -130,97 +211,93 @@ export default function useAuthentication() {
+  /**
+   * Gets the google user's auth token and signs in using the auth plugin
+   * @param idToken: The auth code returned by Google after a sucessful sign in
+   */
+  async function authenticateWithGoogle({ idToken }) {
+    setLoading(true);
 
-    const scope = "https://www.googleapis.com/auth/userinfo.email openid";
+    const data = await authenticateGoogle({
+      idToken,
+      pendingGroup,
+    });
 
-    const responseType = "code";
+    const returnRole =
+      pendingGroup === "teachers"
+        ? "googleSignInTeachers"
+        : "googleSignInStudents";
 
-    return `https://accounts.google.com/o/oauth2/v2/auth?scope=${scope}&client_id=${GOOGLE_CLIENT_ID}&response_type=${responseType}&redirect_uri=${GOOGLE_REDIRECT_URI}`;
+    return data?.[returnRole]
+      ? handleSuccess(data[returnRole])
+      : handleError(data);
   }
 
-  /** Gets the google user's auth token and signs in using the auth plugin
-   * code: The auth code returned by Google after a sucessful sign in
-   * role: The user's role (student or teacher)
+  /**
+   * 1. Get the FB sign-in URL from Craft
+   * 2. Redirect to FB sign-in
+   * 3. After user is authenticated by FB, they will be redirected back
+   *    to this site with a `code` URL param.
+   * 4. An effect above watches for changes to URL params and
+   *    executes `authenticateWithFacebook` when `code` and `facebook` params are present
    */
-  async function authenticateWithGoogle({ code, role }) {
-    const tokenData = await getGoogleOauthToken({ code, role });
-
-    if (tokenData.id_token) {
-      console.info("Authenticating with Google...");
-      const data = await authenticateGoogle({
-        token: tokenData.id_token,
-        role,
-      });
-
-      const returnRole =
-        role === "teacher" ? "googleSignInTeachers" : "googleSignInStudents";
+  async function goToFacebookSignIn() {
+    setLoading(true);
 
-      if (data?.[returnRole]) {
-        setStateFromResponse(data[returnRole]);
-      }
+    const data = await getFacebookOauthUrl();
 
-      return data;
+    if (data?.facebookOauthUrl) {
+      window.open(data.facebookOauthUrl, "_self");
     } else {
-      // TODO: Would want to show an error or return an error...
-      return tokenData;
+      handleError(data);
     }
   }
 
-  async function getFacebookAuthUrl() {
-    const data = await getFacebookOauthUrl();
-
-    return data;
-  }
+  async function authenticateWithFacebook({ code }) {
+    setLoading(true);
 
-  async function authenticateWithFacebook({ code, role }) {
-    const data = await authenticateFacebook({ code, role });
+    const data = await authenticateFacebook({ code, pendingGroup });
 
     const returnRole =
-      role === "teacher" ? "googleSignInTeachers" : "googleSignInStudents";
-
-    if (data?.[returnRole]) {
-      setStateFromResponse(data[returnRole]);
-    }
+      pendingGroup === "teachers"
+        ? "facebookSignInTeachers"
```

Comment:
When I try to register as a teacher, I'm getting this error: 
```
"Cannot query field "facebookSignInTeachers" on type "Mutation". Did you mean "facebookSignInEditors", "facebookSignInEducators", or "facebookSignInStudents"?"
```
If the schema has changed to "Editors" then the google & facebook mutations will need to update too.

I also had trouble signing in as a student with facebook/google:
```
"No schema has been set for this user group"
```
<img width="620" alt="Screen Shot 2022-03-16 at 1 31 51 PM" src="https://user-images.githubusercontent.com/57107444/158685579-19336e8c-6e71-471e-b1ba-774a060c3162.png">